orb/core/collection.py

```diff
@@ -302,7 +321,46 @@ def first(self, **context):
                     self.__cache['first'][context] = record
                 return record
 
-    def hasRecord(self, record, **context):
+    def grouped(self, *columns, **context):
+        preload = context.pop('preload', False)
```

Comment:
It's explained in the docs, one of the things about Python when you start using the magic, you can't mix and match args and kwds with pre-defined ones:

```
def grouped(self, *columns, preload=False, **context):
```

preload will load all records, and then group them.  not preloading will only load the values that you want to group by, and then create collections that can query a subset.

https://orb-framework.gitbooks.io/orb/content/api-reference/collections.html  (look for Grouping)